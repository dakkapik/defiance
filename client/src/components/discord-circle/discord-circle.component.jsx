import React from "react";
import "./discord-circle.styles.scss";
const DiscordCircle = ({ lightSwitch }) => {
  return (
    <div className="discord-circle-container">
      {lightSwitch ? (
        <div className="online-circle " />
      ) : (
        <div className="offline-circle" />
      )}
    </div>
  );
};

export default DiscordCircle;
