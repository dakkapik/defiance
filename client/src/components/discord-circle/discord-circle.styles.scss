.discord-circle-container {
  margin-left: 33px;
  position: relative;
  top: -5px;
}
.online-circle {
  background: rgb(5, 243, 191);
  border-radius: 50%;
  height: 8px;
  width: 8px;
  position: relative;
  box-shadow: 0 0 0 3px black;
  //   opacity: 0.3;
  transition: 0.3s;
}
.offline-circle {
  background: rgb(78, 71, 71);
  border-radius: 50%;
  height: 8px;
  width: 8px;
  position: relative;
  box-shadow: 0 0 0 3px black;
  //   opacity: 0.3;
  transition: 0.3s;
}
