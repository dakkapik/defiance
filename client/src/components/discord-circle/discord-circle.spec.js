import React from "react";
import { DiscordCircle } from "./discord-circle.component";
import { shallow } from "enzyme";

describe("DiscordCircle component render", () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      lightSwitch: true,
    };
    wrapper = shallow(<DiscordCircle {...mockProps} />);
  });

  it("should render DiscordCircle component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
