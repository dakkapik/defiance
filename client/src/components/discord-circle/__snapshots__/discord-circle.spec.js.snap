// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscordCircle component render should render DiscordCircle component 1`] = `ShallowWrapper {}`;
